### PR TITLE
add global serverHostname_g

### PR DESCRIPTION
**Pull request by mxafi**
## Why get the hostname?

Some IRC replies require sending the server hostname as the prefix before the command, or as a parameter after the command.

## Why use a global?

The serverHostname_g variable is going to be used in IRC numeric replies, which are going to be implemented as pre-processor macros.
Those macros will be used in the Client class, which is not yet implemented.
The macros will possibly be used in other classes as well, that is unclear at this point in time.

Requiring Clients to depend on the Server class (by adding the variable to the Server class) would introduce a circular dependancy, as the Server class will have to depend on the Client class (Server contains Clients).

In conclusion, the global is useful because the variable would have been implemented as a fake global to be passed around anyway. If that is not the case, the variable and the functions related to it can be later modified to be private to the only class that requires them.

## Behaviour

The global variable serverHostname_g contains the hostname of the server, which is set by the operating system.
In the case that the hostname is over 63 characters long, the variable contains the host address instead. (e.g. 0.0.0.0 or 127.0.0.1 or 194.136.126.51)

Server hostname is used for 63 characters and shorter hostnames and in longer ones the host address is used instead.
This behaviour is due to the IRC [RFC 2812 section 2.3.1 Message format in Augmented BNF](https://datatracker.ietf.org/doc/html/rfc2812#section-2.3.1):
```
. . . (OMITTED)
      2) <hostname> has a maximum length of 63 characters.  This is a
         limitation of the protocol as internet hostnames (in
         particular) can be longer.  Such restriction is necessary
         because IRC messages are limited to 512 characters in length.
         Clients connecting from a host which name is longer than 63
         characters are registered using the host (numeric) address
         instead of the host name.
. . . (OMITTED)
```

### CODE REVIEW

File: src/main/main.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "main.h"  // IWYU pragma: keep // due to convention
 
 bool isServerRunning_g = true;
+std::string serverHostname_g;
```
Comment (mxafi) [inline src/main/main.cpp:4]: string is appropriate due to being c++ and getting automatically deleted when out of scope

File: src/server/Server.cpp [hunk 2]
```diff
@@ -8,18 +8,49 @@ irc::Server::~Server() {
...
+
 int irc::Server::start() {
   memset(&hints_, 0, sizeof hints_);
   hints_.ai_family = server_socket_domain_;
   hints_.ai_socktype = server_socket_type_;
   hints_.ai_flags = AI_PASSIVE;
   if (int gai_ret = getaddrinfo(NULL, port_, &hints_, &srvinfo_) != SUCCESS) {
     LOG_ERROR("server getaddrinfo failed: (" << gai_ret << ") "
-                                               << gai_strerror(gai_ret));
+                                             << gai_strerror(gai_ret));
```
Comment (mxafi) [inline src/server/Server.cpp:44]: Automatic formatting

File: src/server/Server.cpp [hunk 3]
```diff
@@ -81,9 +112,11 @@ void irc::Server::loop() {
         } else {
           // handle incoming request from existing client connection
         }
-      } if (it->revents & POLLOUT) {  // ready to send()
+      }
+      if (it->revents & POLLOUT) {  // ready to send()
         // handle outgoing response to existing client connection
-      } if (it->revents & POLLERR) {  // socket disconnect or error
+      }
+      if (it->revents & POLLERR) {  // socket disconnect or error
```
Comment (mxafi) [inline src/server/Server.cpp:119]: automatic formatting

File: src/server/Server.h [hunk 4]
```diff
@@ -1,6 +1,7 @@
 #ifndef SERVER_H
 #define SERVER_H
 
+#include <arpa/inet.h>
```
Comment (mxafi) [inline src/server/Server.h:4]: necessary for inet_ntop()

**Review by djames9 — APPROVED** [review]
LOG_ERROR("server inet_ntop failed: " << strerror(errno)); // i think we are not allow to use errno is according to cases.thanks

**Comment by mxafi** [comment]
>  LOG_ERROR("server inet_ntop failed: " << strerror(errno)); // i think we are not allow to use errno is according to cases.thanks 

From devdocs.io:
```
errno is a preprocessor macro used for error indication. It expands to a static(until C++11)thread-local(since C++11) modifiable lvalue of type int.
```

The subject only prohibits the use of Boost and external libraries in my interpretation.

Which part of the subject are you referring to?
@djames9

**Review by liocle — APPROVED** [review]
Looks good to me after the discussion with Diego is resolved regarding errno.

**Pull request merged**